tests/components/drop_connect/test_coordinator.py

```diff
@@ -1,74 +1,65 @@
 """Test DROP coordinator."""
-from homeassistant.components.drop_connect.const import DOMAIN
 from homeassistant.config_entries import ConfigEntryState
 from homeassistant.const import STATE_UNAVAILABLE, STATE_UNKNOWN
 from homeassistant.core import HomeAssistant
-from homeassistant.setup import async_setup_component
 
-from .common import TEST_DATA_HUB, TEST_DATA_HUB_RESET, TEST_DATA_HUB_TOPIC
+from .common import (
+    TEST_DATA_HUB,
+    TEST_DATA_HUB_RESET,
+    TEST_DATA_HUB_TOPIC,
+    config_entry_hub,
+)
 
 from tests.common import async_fire_mqtt_message
 from tests.typing import MqttMockHAClient
 
 
-async def test_bad_json(
-    hass: HomeAssistant, config_entry_hub, mqtt_mock: MqttMockHAClient
-) -> None:
+async def test_bad_json(hass: HomeAssistant, mqtt_mock: MqttMockHAClient) -> None:
     """Test bad JSON."""
-    config_entry_hub.add_to_hass(hass)
-    assert await async_setup_component(hass, DOMAIN, {})
-    await hass.async_block_till_done()
+    entry = config_entry_hub()
+    entry.add_to_hass(hass)
+    assert await hass.config_entries.async_setup(entry.entry_id)
 
     current_flow_sensor_name = "sensor.hub_drop_1_c0ffee_water_flow_rate"
-    hass.states.async_set(current_flow_sensor_name, STATE_UNKNOWN)
+    assert hass.states.get(current_flow_sensor_name).state == STATE_UNKNOWN
 
     async_fire_mqtt_message(hass, TEST_DATA_HUB_TOPIC, "{BAD JSON}")
     await hass.async_block_till_done()
-
-    current_flow_sensor = hass.states.get(current_flow_sensor_name)
-    assert current_flow_sensor
-    assert current_flow_sensor.state == STATE_UNKNOWN
+    assert hass.states.get(current_flow_sensor_name).state == STATE_UNKNOWN
 
 
-async def test_unload(
-    hass: HomeAssistant, config_entry_hub, mqtt_mock: MqttMockHAClient
-) -> None:
+async def test_unload(hass: HomeAssistant, mqtt_mock: MqttMockHAClient) -> None:
```

Comment:
I think this test can be moved to a `test_init.py` in a followup. And we don't really have to test the states, just if it can load and unload